ast.go

```diff
@@ -3598,6 +3769,9 @@ type Distinct struct {
 
 // String returns a string representation of the expression.
 func (d *Distinct) String() string {
+	if d == nil {
+		return ""
+	}
```

Comment:
Check `d.Val` for nil?

Reply:
Nope, `d.Val` is a non-pointer string type